src/Integration.Vsix/SonarLintDaemon/SonarLintDaemonInstaller.xaml.cs

```diff
@@ -32,15 +31,27 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </summary>
     public partial class SonarLintDaemonInstaller : Window
     {
-        private ISonarLintSettings settings;
-        private ISonarLintDaemon daemon;
+        private readonly ISonarLintSettings settings;
+        private readonly ISonarLintDaemon daemon;
 
         private volatile bool canceled = false;
 
         private Action callback;
 
-        public SonarLintDaemonInstaller()
+        public SonarLintDaemonInstaller(ISonarLintSettings settings, ISonarLintDaemon daemon)
```

Comment:
Removing the use of _ServiceProvider.GlobalProvider_ means these objects need to be supplied by the caller (in this case it's the package, which has both already).